go-controller/pkg/cni/helper_linux.go

```diff
@@ -411,21 +474,11 @@ func (pr *PodRequest) deletePodConntrack() {
 	}
 }
 
-func (pr *PodRequest) deletePorts() {
-	ifaceName := pr.SandboxID[:15]
+func (pr *PodRequest) deletePorts(ifaceName string) {
 	out, err := ovsExec("del-port", "br-int", ifaceName)
 	_ = util.LinkDelete(ifaceName)
```

Comment:
This would delete VF rep port when ovs failed to configure. Deleted VF rep is no longer visible in host namespace.
Maybe add conditional check to not execute linkDelete for CX5 case.

Reply:
@zshi-redhat  we cannot delete a VF rep port as can be seen below

```
# ethtool -i ens4f1_1
driver: mlx5e_rep
version: 4.18.0-193.13.2.el8.1623039296.
firmware-version: 16.31.1014 (MT_0000000080)
expansion-rom-version: 
bus-info: 
supports-statistics: yes
supports-test: no
supports-eeprom-access: no
supports-register-dump: no
supports-priv-flags: no
# ip li del  ens4f1_1
RTNETLINK answers: Operation not supported
```

Reply:
Let me double check, I remembered I have hit issue that VF rep is not found in the host netns, I suspected that it was due to this LinkDelete call.  On the other hand, I think we also want to avoid such failure on deleting VF rep ports.

Reply:
Girirsh is right, you can delete the VF rep netdevie.

Reply:
Can or cannot ?
I tried manually, it cannot (unsure why the vf rep disappear from hostnamespace in previous testing).

Reply:
Even it cannot be deleted, I think we should avoid executing the deletion for vf rep device (it would fail and report error).

Reply:
Sorry I meant can't :)

Reply:
@moshe010 @girishmg  We noticed an error message that vf rep device was not found from ovs-vswitchd log when running ovs hwol performance testing (delete and recreate pods multiple times will likely trigger this issue):

```
Dec 02 18:19:04 worker-2 ovs-vswitchd[9550]: ovs|01771|netdev_offload_tc(revalidator49)|ERR|dump_create: failed to get ifindex for 97d3304f5d35a02: No such device
```
I think this is the one I have seen before, maybe it is not due to this LinkDelete call in ovn-k8s, did you ever hit such issue ?